roles/confluent.ksql/tasks/rbac.yml

```
@@ -144,3 +148,32 @@
       }
     status_code: 204
   when: "'schema_registry' in groups"
+
+- name: Grant ksql user the ResourceOwner role on the Processing Log Topic
+  uri:
+    url: "{{mds_http_protocol}}://{{ groups['kafka_broker'][0] }}:{{mds_port}}/security/1.0/principals/User:{{ksql_ldap_user}}/roles/ResourceOwner/bindings"
```

```suggestion
    url: "{{mds_http_protocol}}://{{ groups['kafka_broker'][0] }}:{{mds_port}}/security/1.0/principals/User:{{ksql_log4j_principal}}/roles/ResourceOwner/bindings"
```

is this supposed to be the log4j principal or the ksql ldap user?

---

Yep, but I keep having to re-edit things due to previous changes, and my changes keeping getting munged.  If you notice it was correct previously.  Honestly, I'm not sure why this keeps happening.  I fix something and then something else gets committed.  I've been using stash's and I think that's where the confusion is coming from.

---

So I figured out what is happening.  You need to grant Resource Owner on the log streaming topic to both users.

The ksql_ldap_user, is the user KSQL runs as via it's OAuth connection, and is what is used to create the topic when ksql is started and log streaming is enabled.

The ksql_log4j_principal, is the user we extract from the keytab in this instance, or TLS cert etc, that is used to create the client connection and write to the log streaming topic, thus it also needs to be a resource owner.

So both users need these permissions when log streaming is enabled.

I've tested this by only having the ksql_log4j_principal set, and KSQL fails to start, because it is unable to create the Topic, it actually throws an error stating it can't authorize to create it.

TL:DR. You need both users to have access to the log streaming topic, when log streaming is enabled.

---

copy that! nice investigation